File: problem-solving/algorithms/breaking-the-records/breaking_the_records.py
def breakingRecords(scores):
    max_breaks, min_breaks = 0, 0

    max_record, min_record = scores[0], scores[0]

    for score in scores[1:]:
        if score > max_record:
            max_breaks += 1
            max_record = score

        if score < min_record:
            min_breaks += 1
            min_record = score

    return [max_breaks, min_breaks]

File: problem-solving/algorithms/breaking-the-records/breaking-the-records.js
function breakingRecords(scores) {
    let maxBreaks = 0, minBreaks = 0;
    let maxRecord = scores[0], minRecord = scores[0];

    for (let i = 1; i < scores.length; i++) {
        if (scores[i] > maxRecord) {
            maxBreaks++;
            maxRecord = scores[i];
        }

        if (scores[i] < minRecord) {
            minBreaks++;
            minRecord = scores[i];
        }
    }

    return [maxBreaks, minBreaks];
}
